code-search/code-search.ipynb

```diff
@@ -0,0 +1,609 @@
+      "source": [
+        "hits = client.search(\n",
+        "    \"qdrant-sources\",\n",
+        "    query_vector=(\n",
+        "        \"code\", code_model.encode(query).tolist()\n",
+        "    ),\n",
+        "    limit=5,\n",
+        ")\n",
+        "for hit in hits:\n",
+        "    print(\n",
+        "        \"| \",\n",
+        "        hit.payload[\"context\"][\"module\"], \" | \",\n",
+        "        hit.payload[\"context\"][\"file_name\"], \" | \",\n",
+        "        hit.score, \" | `\",\n",
+        "        hit.payload[\"signature\"], \"` |\"\n",
+        "    )"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "In reality, we implemented the system with two different models, as we want to combine the results coming from both of them. We can do it with a batch request, so there is just a single call to Qdrant."
+      ],
+      "metadata": {
+        "id": "It9JqvXDVN17"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "xh68mAfCsfQW"
+      },
+      "outputs": [],
+      "source": [
+        "results = client.search_batch(\n",
+        "    \"qdrant-sources\",\n",
+        "    requests=[\n",
+        "        models.SearchRequest(\n",
+        "            vector=models.NamedVector(\n",
+        "                name=\"text\",\n",
+        "                vector=nlp_model.encode(textify(query)).tolist()\n",
+        "            ),\n",
+        "            with_payload=True,\n",
+        "            limit=5,\n",
+        "        ),\n",
+        "        models.SearchRequest(\n",
+        "            vector=models.NamedVector(\n",
+        "                name=\"code\",\n",
+        "                vector=code_model.encode(query).tolist()\n",
+        "            ),\n",
+        "            with_payload=True,\n",
+        "            limit=5,\n",
+        "        ),\n",
+        "    ]\n",
+        ")\n",
+        "for hits in results:\n",
+        "    for hit in hits:\n",
+        "        print(\n",
+        "            \"| \",\n",
+        "            hit.payload[\"context\"][\"module\"], \" | \",\n",
+        "            hit.payload[\"context\"][\"file_name\"], \" | \",\n",
+        "            hit.score, \" | `\",\n",
+        "            hit.payload[\"signature\"], \"` |\"\n",
+        "        )"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Last but not least, if we want to improve the diversity of the results, grouping them by the module might be a good idea."
+      ],
+      "metadata": {
+        "id": "x2Fmp22CVZsi"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "results = client.search_groups(\n",
+        "    \"qdrant-sources\",\n",
+        "    query_vector=(\n",
+        "        \"code\", code_model.encode(query).tolist()\n",
+        "    ),\n",
+        "    group_by=\"context.module\",\n",
```

Comment:
This looks excellent! 

Do you want to show inline how the output might look like for this? E.g. will it be two lists or one? Are groups just lists and so on?

Reply:
It actually makes sense to add all the outputs. I'll run the notebook and upload a version with outputs.